app/elements/io-session-details.html

```diff
@@ -302,8 +302,23 @@ <h5 class="session__time">May [[selectedSession.day]], [[selectedSession.start]]
       }
     },
 
-    sessionHasVideo: function(startVideo, isLivestream, youtubeUrl) {
-      return startVideo && !isLivestream && youtubeUrl;
+    sessionHasVideo: function(startVideo, selectedSession) {
+      if (!selectedSession) {
+        return false;
+      }
+
+      var hasStarted = new Date(selectedSession.startTimestamp) < Date.now();
+      var hasEnded =  Date.now() > new Date(selectedSession.endTimestamp);
+
+      var duringLiveSession = selectedSession.isLivestream &&
+                              selectedSession.youtubeUrl &&
+                              hasStarted && !hasEnded;
+
+      // Show player if it's during a livestream session or the video has
+      // been posted. When the video is posted, the youtubeUrl becomes a full
+      // youtube URL and isLivestream is false.
+      return startVideo && !selectedSession.isLivestream &&
```

Comment:
Yea. The first part of this `||` is for video case:
"When the video is posted, the youtubeUrl becomes a full  youtube URL and isLivestream is false."